### ISSUE TITLE
Client does not connect to server [Wireguard instructions]

### ISSUE THREAD

**Issue by g0djira** [erebe/wstunnel#237]
**Describe the bug**
As described in [Wireguard and wstunnel] section I've started 
- on server:
```
$wstunnel server --restrict-to localhost:51510 wss://[::]:8080
2024-03-07T15:30:50.003898Z  INFO wstunnel::embedded_certificate: Loading embedded tls certificate
2024-03-07T15:30:50.004397Z  INFO wstunnel::embedded_certificate: Loading embedded tls private key
2024-03-07T15:30:50.005180Z  INFO wstunnel: Starting wstunnel server v9.2.2 with config WsServerConfig { socket_so_mark: None, bind: [::]:8080, restrict_to: Some(["localhost:51510"]), restrict_http_upgrade_path_prefix: None, websocket_ping_frequency: None, timeout_connect: 10s, websocket_mask_frame: false, tls: true }
2024-03-07T15:30:50.005685Z  INFO wstunnel::tunnel::server: Starting wstunnel server listening on [::]:8080
```
- on client:
```
$wstunnel client -L 'udp://51510:localhost:51510?timeout_sec=0' wss://my.server:8080                                                                                   
2024-03-07T15:31:51.736283Z  INFO wstunnel::udp: Starting UDP server listening cnx on 127.0.0.1:51510 with cnx timeout of 0s   
```

And that's all, I can't see in logs that client is trying to connect to server and exchange certificates.

**To Reproduce**
1. Start server with given parameters.
2. Start client with given parameters.

**Expected behavior**
- server accepting connection;
- server and client doing tls handshake.

**Screenshots**

**Desktop (please complete the following information):**
 - server: Ubuntu 22.04.4 LTS
 - client: Raspbian GNU/Linux 11 (bullseye)
 - Version 9.2.3

**Additional context**
wstunnel works as expected in example "Simplest one"

**Comment by erebe** [comment]
Hello,

Most likely, you misconfigured your WireGuard client configuration. Be sure, the endpoint point to localhost, and to start wireguard after wstunnel is started.

**Comment by g0djira** [comment]
Fixed by changing `localhost` to `127.0.0.1` in [Peer] section, client's wg config. O_o

**Comment by erebe** [comment]
Perfect then, i assume localhost in your /etc/hosts is set properly ?
